internal/rpc/rpcserver/server.go

```diff
@@ -3312,11 +3306,14 @@ func marshalDecodedTxOutputs(mtx *wire.MsgTx, chainParams *chaincfg.Params) []*p
 		// accordingly.
 		var addrs []stdaddr.Address
 		var encodedAddrs []string
-		var scriptClass txscript.ScriptClass
-		var reqSigs int
+		var scriptClass stdscript.ScriptType
+		var reqSigs uint16
 		var commitAmt *dcrutil.Amount
-		if (txType == stake.TxTypeSStx) && (stake.IsStakeSubmissionTxOut(i)) {
-			scriptClass = txscript.StakeSubmissionTy
+		if (txType == stake.TxTypeSStx) && (stake.IsStakeCommitmentTxOut(i)) {
+			// Questionable! This is either "nulldata" or the pseudo-type
+			// "sstxcommitment", not "stakesubmission", which is only the 0th
+			// output of a ticket.
+			scriptClass = stdscript.STStakeSubmissionPubKeyHash
```

Comment:
I'd like to resolve this bug now, but some discussion is needed with @alexlyp and @matheusd, and any other consumers of the `MessageService` and `DecodeRawTransaction` in particular.

For background, this function was added for Decrediton (https://github.com/decred/dcrwallet/pull/981), but it may be used elsewhere.  The functions were modeled after the JSON-RPC handlers it seems, but I think due to the original naming of `stake.IsStakeSubmissionTxOut`, which is now `stake.IsStakeCommitmentTxOut`, the `script_class` was set to `STAKE_SUBMISSION` instead of either `NULL_DATA` or a new custom pseudo-type to indicate a stake commitment.

For example https://testnet.dcrdata.org/tx/449a8f490a85558198e8c3dc3911b336df82a3f069fda009cf0a6073bd4fec39, only the 0th output should be `STAKE_SUBMISSION`:

```
transaction_hash: "9\xecO\xbds`\n\xcf\t\xa0\xfdi\xf0\xa3\x82\xdf6\xb3\x119\xdc\xc3蘁U\x85\nI\x8f\x9aD"
version: 1
expiry: 815760
transaction_type: TICKET_PURCHASE
inputs: {
  previous_transaction_hash: "m\xb5\x9a\x8f\x8e\x1c\x9d\xb0\xa9c\xb0\x8a0\xf0/\x1c\xea\x95\xdd\xd400:\x8d\x98Y\xees\x0c\x01\xd8e"
  sequence: 4294967295
  amount_in: 10243989325
  block_height: 2147483647
  block_index: 4294967295
  signature_script: "H0E\x02!\x00\xe0K\x03\xb2\x84J\xde*\xb8HG\x90-\xac<\xcd`\"\xbc#\xdc\x01*\x119\nY\x04lY\x18i\x02 ,\x0fk\xdd\xfaKL$\x94\xf4\x0b\xfc\x8a~\xc0z\xb5pľmf\xfa\x10\xb4\x85U޶aH\xd1\x01!\x02H\xbe]%\x01\xe3\xa9\xed0y\x13\x8aH\xe9C\xc1.x\xdaV\xedu,\xafY\x0cQ\x0f\x1c\xe3\xc9)"
  signature_script_asm: "3045022100e04b03b2844ade2ab84847902dac3ccd6022bc23dc012a11390a59046c59186902202c0f6bddfa4b4c2494f40bfc8a7ec07ab570c4be6d66fa10b48555deb66148d101 0248be5d2501e3a9ed3079138a48e943c12e78da56ed752caf590c510f1ce3c929"
}
outputs: {
  value: 10243986345
  script: "\xbav\xa9\x14\xfd^ \x12\x8a.\\K٫\xf93\x86\x91+byQ\xe5Ĉ\xac"
  script_asm: "OP_SSTX OP_DUP OP_HASH160 fd5e20128a2e5c4bd9abf93386912b627951e5c4 OP_EQUALVERIFY OP_CHECKSIG"
  required_signatures: 1
  script_class: STAKE_SUBMISSION
  addresses: "Tsp7pC2pzeTowrGzhHnopdMDjhQL3tosVdb"
}
outputs: {
  index: 1
  script: "j\x1e\x89\x9e\xb2>ӵ\x1c\xfe\x83\x9fp\x86x\xaf\x83=\xaa_(yMߖb\x02\x00\x00\x00\x00N"
  script_asm: "OP_RETURN 899eb23ed3b51cfe839f708678af833daa5f28794ddf966202000000004e"
  script_class: STAKE_SUBMISSION
  addresses: "TsdZoAFWL4PUgfifbdrcPqCStaA2VFo6fBe"
}
outputs: {
  index: 2
  script: "\xbdv\xa9\x14\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x88\xac"
  script_asm: "OP_SSTXCHANGE OP_DUP OP_HASH160 0000000000000000000000000000000000000000 OP_EQUALVERIFY OP_CHECKSIG"
  required_signatures: 1
  script_class: STAKE_SUB_CHANGE
  addresses: "TsR28UZRprhgQQhzWns2M6cAwchrNVvbYq2"
}
```

The question now is, are there consumers that rely on seeing the stake commitment outputs (the special odd-numbered outputs that have op_return pushes encoding commitment address, amount, and fee info) that would break if this is fixed?

The second question is, if it does get fixed, should we have these be `NULL_DATA` or a new `STAKE_COMMITMENT` `script_class`?  `NULL_DATA` is easiest, but it's also straightforward to add a new class to api.proto and change how we prepare the value for the `pb.DecodedTransaction_Output`.  It'll only take a few minutes.

Reply:
Leaving existing behavior.  The misidentified stake commitment script class can be fixed in another PR.